### PR TITLE
Start creating a new plan from the list of plans

### PR DESCRIPTION
**Pull request by savannidgerinel**
Add the modal dialogue that allows the creation of a new plan directly from the list of plans.

In additional to making the modal work, I decomposed the behavior of the assessment selection actions so that all of the selection modals can use the same controller.

IMPORTANT: this impacts the landing page, the Goals page, and the List of Plans page. Be sure to test country selection and plan creation process on all three.

# Screenshots
Screen Shot 2019-08-29 at 5 48 51 PM
Screen Shot 2019-08-29 at 5 49 01 PM

# Stories
Resolves: https://www.pivotaltracker.com/story/show/167604994

### CODE REVIEW

File: app/views/shared/_joint_assessment_selection_modal.html.erb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        </p>
+
+        <p>
+          If the country hasn’t done either assessment, you can create a plan
+          from technical areas.
+        </p>
+
+        <%= form_tag '/country_select', id: "assessment-select-menu" do %>
+          <%= hidden_field_tag "country", nil, data: { target: "assessment-selection.selectedCountryName" } %>
+          <div class="input-group">
```
Comment (sidk) [inline app/views/shared/_joint_assessment_selection_modal.html.erb]: this `div` needs a closing tag - i see layout issues with the footer without it.

File: app/views/shared/_joint_assessment_selection_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          the capacity score and step up score.
+        </p>
+
+        <p>
+          You can create a plan from the Joint External Evaluation (JEE) or
+          State Party Self-Assessment Annual Reporting (SPAR) scores.
+        </p>
+
+        <p>
+          If the country hasn’t done either assessment, you can create a plan
```
[the remark this answers is not in the document]
Comment (savannidgerinel) [reply app/views/shared/_joint_assessment_selection_modal.html.erb]: It really isn't accurate, and I think I should mention it to Stacy.

[the remark this answers is not in the document]
Comment (savannidgerinel) [reply app/views/shared/_joint_assessment_selection_modal.html.erb]: Filed as https://www.pivotaltracker.com/story/show/168222920

File: app/javascript/src/controllers/joint_country_select_controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Controller } from "stimulus"
```
[the remark this answers is not in the document]
Comment (savannidgerinel) [reply app/javascript/src/controllers/joint_country_select_controller.js]: Done.

**Pull request merged**